torch/csrc/jit/codegen/cuda/ir_base_nodes.cpp

```diff
@@ -144,6 +144,23 @@ Expr* Val::getOrigin() const {
   return fusion_->origin(this);
 }
 
+bool Val::isProducer(const Val* other) const {
```

Comment:
Nit, maybe we should name this as:
isConsumerOf(other)
and
isProducerOf(other)

isConsumer/Producer isn't an inherent property of Val, but a relation between two vals, the syntax just seems a little more clear to me, but don't have a strong opinion of you object

Reply:
Thanks. Makes sense. Updated.